src/main/java/org/assertj/core/internal/Files.java

```diff
@@ -88,10 +89,15 @@ public static Files instance() {
   public void assertSameContentAs(AssertionInfo info, File actual, File expected) {
     verifyIsFile(expected);
     assertIsFile(info, actual);
+    BinaryDiffResult binaryDiffResult = null;
     try {
+      binaryDiffResult = binaryDiff.diff(actual, java.nio.file.Files.readAllBytes(expected.toPath()));
+      if (binaryDiffResult.hasNoDiff()) return;
       List<Delta<String>> diffs = diff.diff(actual, expected);
       if (diffs.isEmpty()) return;
       throw failures.failure(info, shouldHaveSameContent(actual, expected, diffs));
+    } catch (MalformedInputException e) {
+      throw failures.failure(info, shouldHaveBinaryContent(actual, binaryDiffResult));
```

Comment:
I don't think it is correct to return an `shouldHaveBinaryContent` failure, it we can't read the file you can't say anything about its content, I think it's should be treated as an `IOException`. WDYT ?